tracing/traced_bcrypt.go

```diff
@@ -0,0 +1,47 @@
+package tracing
+
+import (
+	"context"
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+
+	"github.com/pkg/errors"
+	"golang.org/x/crypto/bcrypt"
+)
+
+const (
+	HashOpName        = "bcrypt.hash"
+	CompareOpName     = "bcrypt.compare"
+	WorkFactorTagName = "bcrypt.workfactor"
+)
+
+// TracedBCrypt implements the Hasher interface
+type TracedBCrypt struct {
+	WorkFactor int
+}
+
+func (b *TracedBCrypt) Hash(ctx context.Context, data []byte) ([]byte, error) {
+	span, _ := opentracing.StartSpanFromContext(ctx, HashOpName)
```

Comment:
Note: the second return value here is a context object built around the returned span which you can use for further propagation. Not applicable in this case hence the `_`.